### DISCUSSION TITLE
NLOHMANN_DEFINE_TYPE_INTRUSIVE_WITH_DEFAULT

### DISCUSSION THREAD

**Discussion by demacia666666** [nlohmann/json#4395] [General]
`#include <iostream>
#include <vector>
#include <string>
#include <optional>
#include "json.hpp"
using json = nlohmann::json;
class User {
public:
    std::string name = {};
    NLOHMANN_DEFINE_TYPE_INTRUSIVE_WITH_DEFAULT(User, name)
};

int main() {
    std::string jsonString = u8R"({"name": null})";
    try {
        /*
       I would prefer a default value, 
       although it can be solved using ADL with std::optional.
       Is there a more elegant way? 
        */
        json jsonData = json::parse(jsonString);
        User user = jsonData;
    }
    catch (const std::exception& e) {
        //[json.exception.type_error.302] type must be string, but is null
        std::cerr << "" << e.what() << std::endl;
    }
    return 0;
}
`

**Comment by gregmarr** [comment]
You need to enclose your code in with triple backticks to get appropriate formatting and to be able to see the `<>` includes.

**Comment by gregmarr** [comment]
The issue is the `null` in the JSON.  The default is only for "not specified", not "specified but with a null value".
